coderd/workspaces.go

```diff
@@ -628,6 +631,18 @@ func (api *API) patchWorkspace(rw http.ResponseWriter, r *http.Request) {
 	// patched in the future, it's enough if one changes.
 	name := workspace.Name
 	if req.Name != "" || req.Name != workspace.Name {
+		if !api.Options.AllowWorkspaceRenames {
+			httpapi.Write(ctx, rw, http.StatusBadRequest, codersdk.Response{
+				Message: "Workspace renames are not allowed.",
+			})
+			return
+		}
+		if api.Options.AllowWorkspaceRenamesExpiresAt.Before(time.Now()) {
```

Comment:
This code could be cleaned up to a single check if we kept ExpiresAt at the cli.